Assets/MRTK/Services/CameraSystem/ProjectionOverride.cs

```diff
@@ -0,0 +1,121 @@
+            {
+                return;
+            }
+
+            StartCoroutine(ResetViewMatricesOnFrameEnd());
+
+            IntPtr nativeStruct = UnityEngine.XR.XRDevice.GetNativePtr();
+            if (nativeStruct != IntPtr.Zero)
+            {
+                HolographicFrameNativeData s = System.Runtime.InteropServices.Marshal.PtrToStructure<HolographicFrameNativeData>(nativeStruct);
+
+                if (s.IHolographicFramePtr != IntPtr.Zero)
+                {
+                    var holographicCamera = (HolographicCamera)System.Runtime.InteropServices.Marshal.GetObjectForIUnknown(s.IHolographicCameraPtr);
+                    var holographicFrame = (HolographicFrame)System.Runtime.InteropServices.Marshal.GetObjectForIUnknown(s.IHolographicFramePtr);
+                    HolographicFramePrediction prediction = holographicFrame.CurrentPrediction;
+
+                    if (holographicCamera.CanOverrideViewport)
+                    {
+                        for (int i = 0; i < prediction.CameraPoses.Count; ++i)
+                        {
+                            HolographicCameraPose cameraPose = prediction.CameraPoses[i];
+
+                            // Default spatial locator is the HMD
+                            SpatialLocator defaultSpatialLocator = SpatialLocator.GetDefault();
+
+                            SpatialLocatorAttachedFrameOfReference attachedFrameOfReference = defaultSpatialLocator.CreateAttachedFrameOfReferenceAtCurrentHeading();
+                            SpatialCoordinateSystem attachedCoordinateFrame = attachedFrameOfReference.GetStationaryCoordinateSystemAtTimestamp(prediction.Timestamp);
+
+                            HolographicStereoTransform stereoProjection = cameraPose.ProjectionTransform;
+                            float ResolutionScale = 45.0f / 33.0f;
+
+
+                            Matrix4x4 leftProj =  CameraCache.Main.GetStereoProjectionMatrix(Camera.StereoscopicEye.Left);
+                            Matrix4x4 rightProj =  CameraCache.Main.GetStereoProjectionMatrix(Camera.StereoscopicEye.Right);
+                            leftProj.m00 *= ResolutionScale;
+                            leftProj.m11 *= ResolutionScale;
+                            rightProj.m00 *= ResolutionScale;
+                            rightProj.m11 *= ResolutionScale;
+                            CameraCache.Main.SetStereoProjectionMatrix(Camera.StereoscopicEye.Left, leftProj);
+                            CameraCache.Main.SetStereoProjectionMatrix(Camera.StereoscopicEye.Right, rightProj);
+
+
+                            stereoProjection.Left.M11 *= ResolutionScale;
+                            stereoProjection.Left.M22 *= ResolutionScale;
+                            stereoProjection.Right.M11 *= ResolutionScale;
+                            stereoProjection.Right.M22 *= ResolutionScale;
+
+                            cameraPose.OverrideProjectionTransform(stereoProjection);
```

Comment:
@KevinKennedy Is there a reason we have to set this on both the Unity camera and the underlying HolographicCameraPose?
Would we expect / want the Unity implementation to call `OverrideProjectionTransform` when the projection matrix is changed?

Reply:
The override of the HolographicCameraPose is so the OS can provide the proper pose/projection information to the LSR subsystem.  The modifying of the unity camera does the actual modifying of the camera.

It may make sense to make Unity handle this override explicitly.  There's the trade off of "this code works now" vs having to have two versions of the code based on which version of unity people are using if we get it built into unity.  I'm not sure if there's a case where it would ever make sense to render with one camera and have Unity use another.  It's possible, as people could do some uncommon compositing techniques by switching the cameras projection matrix mid scene and then going back.  By coupling these two calls together it may prevent people from doing those uncommon techniques.  

It's also worth noting that the OverrideProjectionTransform is (or wasn't at the time I wrote this) implemented in the remoting stack.  I don't know if that changes the decision on if this should go into the unity engine camera implementation or not.